src/frontend/http.rs

```diff
@@ -420,10 +440,24 @@ pub fn filters(
 
     let log = warp::log(module_path!());
 
+    // TODO: The path parameter parsing below is very unergonomic, mostly because warp doesn't
+    // provide better primitives (e.g. `warp::generic::Either` is private outside of the crate).
+    // The logic is to try and match the endpoint with or without the optional prefix for the DB
+    // name, and then map the first parameter in both cases to the same `Option<String>` type, which
+    // lets us use the `unify` filter method.
+
     // Cached read query
     let ctx = context.clone();
-    let cached_read_query_route = warp::path!("q" / String)
```

Comment:
Maybe you could use something like `warp::path!("q" / String)[....].then(uncached_read_write_query)`, `warp::path!("q" / String / String)[...].then(uncached_read_write_query_db)` and the two functions would be thin wrappers that call the actual function with a `Some(dbname)`/`None` respectively? Not sure if that would be cleaner though, since you'd be defining four routes.

Reply:
Yeah, I thought about this and it seemed like a lot more verbose solution.

I'm ok with keeping it as is for now, hopefully simplifying it one day to something like
```rust
let cached_read_query_route = warp::path!("q" / String)
        .or(warp::path!(String / "q" / String))
        .and(warp::get())
        .map(|either| match either {
            Either::A(query_hash) => (None, query_hash),
            Either::B((db_name, query_hash)) => (Some(db_name), query_hash),
        })
        ...
        .then(cached_read_query)
        .map(into_response);
```
conditional on `warp::generic::Either` becoming public (I think it's private only due to an oversight).